src/FlowerShop.Core/Entities/Product.cs

```diff
@@ -24,5 +24,28 @@ public Product(string name, decimal price, string description, ProductType produ
         }
 
         public void SetAvailabilityCount(int newCount) => AvailabilityCount = newCount;
+
+        public void UpdateProductDetails(string name, decimal? price, string description, int? quantity)
+        {
+            if(name != null)
+            {
+                Name = name;
+            }
+
+            if(price != null)
+            {
+                Price = (decimal)price;
```

Comment:
nullable has GetValueOrDefault().  Price = price.GetValueOrDefault() and you can skip the null check as it assigns a default value if null or a normal value if it's not null.

Reply:
GetValueOrDefault  sets value to 0 if the price is null >;)

Reply:
You can pass existing Price to GetValueOrDefault(Price), then the default value will remain as Price. 👀 The way we updated entities by using CurrentValues and SetValues  update by checking the properties which were updated, so there shouldn't be any performance impact as if you change the price to what it was it won't get detected as a changed entity property.